app/main.py

```diff
@@ -12,48 +12,46 @@ def __repr__(self) -> str:
         return f"Distance(km={self.km})"
 
     def __add__(self, other: Distance | int | float) -> Distance:
-        if type(other) == int or type(other) == float:
+        if isinstance(other, (int, float)):
             return Distance(self.km + other)
         return Distance(self.km + other.km)
 
     def __iadd__(self, other: Distance | int | float) -> Distance:
-        if type(other) == int or type(other) == float:
+        if isinstance(other, (int, float)):
             self.km += other
         else:
             self.km += other.km
         return self
 
-    def __mul__(self, other: Distance | int | float) -> Distance | None:
-        if type(other) == int or type(other) == float:
+    def __mul__(self, other: int | float) -> Distance | None:
+        if isinstance(other, (int, float)):
             return Distance(self.km * other)
-        return None
 
-    def __truediv__(self, other: Distance | int | float) -> Distance | None:
-        if type(other) == int or type(other) == float:
+    def __truediv__(self, other: int | float) -> Distance | None:
+        if isinstance(other, (int, float)):
```

Comment:
`other` can not be of type `Distance` in this methods, so we don't need to do a check for type of instance. Also fix return type annotatiohs